### ISSUE TITLE
Sidebar with groups and connections

### ISSUE THREAD

**Issue by emmathesaint** [VShawn/PRemoteM#77]
First of all, I would like to thank you for this great tool. I've tried it and it works perfectly fine. One feature, that IMHO would enhance the user experience is to add a sidebar with the groups and connections for each group. I understand that by doing this it would mean a change in the design of the GUI, but could it be added as an option in the configuration? Once again the app works really good, the resizing and behavior of the RDP connections are blazing fast (I've tried different connection managers).

Once again, I appreciate all your effort to develop this great tool.

Emmanuel.

**Comment by VShawn** [comment]
Hi, thanks for your appreciation 😄, I am glad that you like it.

And sorry for I dont get the purpose and how it looks like of the sidebar.

Can you give more detail and a draft design? it would be of great help to us

BTW, You can draw simply on a screenshot by MSPAINT, or try https://excalidraw.com/

**Comment by majkinetor** [comment]
>  to add a sidebar with the groups and connections for each group

@emmathesaint probably wants the same interface as in mRemoteNG. I find this interface problematic and there is nothing wrong with current approach in PRM, especially since it currently doesn't support groups within groups. 

Groups could be added as tree in the list view tho, this would look alot like mRemoteNg only better and interface would work if nested groups are added (which is not easy to achieve with current one). This is already proposed as #46.

**Comment by emmathesaint** [comment]
Thanks @majkinetor  and apologize for my late reply. The idea is not to copy ideas from other software products as mRemoteNG or similars. My request is to be able to have nested groups. This will help a lot to organize the list of the connections and most of all when you have many servers and clients from different locations.

![2021-02-08 11_29_52-PRemoteM](https://user-images.githubusercontent.com/62189058/107233178-0c5f0f00-6a01-11eb-9c4b-b5bba92226f1.png)

I think this will help a lot with the user experience in terms of organization. I understand that some changes are not easy to develop, but in this case I think it really will improve PRemoteM in terms of use. 

Once again @VShawn  thanks a lot for your effort and great software.

Thanks.

Emmanuel.

**Comment by VShawn** [comment]
ahhh , do you mean  the list like this? 

![](https://user-images.githubusercontent.com/10143738/103165931-9c356900-4858-11eb-9e47-1bc5c7d0ee3a.png)

**Comment by emmathesaint** [comment]
@VShawn yes, exactly! if it's possible it really would be helpful! Once again, thanks a lot!

Emmanuel.

**Comment by majkinetor** [comment]
If you go with this @VShawn keep in mind #73 as it may involve setting it on group level.

**Comment by emmathesaint** [comment]
thanks a lot @majkinetor  and @VShawn ! I really appreciate your help about this request!

Emma.

**Comment by VShawn** [comment]
I am not sure if I will go with this, I can not find any WPF listview implement like https://github.com/VShawn/PRemoteM/issues/77#issuecomment-775229158, and it's hard for me to write a new listview control myself. So let‘s put this into limbo before we can find a proper solution.

**Comment by majkinetor** [comment]
I think its probably the best not to introduce new features that depend on libraries before doing #72 as it may turn out that some of those libs may block you.

**Comment by VShawn** [comment]
> I think its probably the best not to introduce new features that depend on libraries before doing #72 as it may turn out that some of those libs may block you.

totally agree.

**Comment by VShawn** [comment]
since `group` is no longer exists I will close this ticke.

If any new idea on our new Tag system, feel free to open a new ticket.
